app/controllers/katello/api/v2/systems_controller.rb

```diff
@@ -418,11 +419,11 @@ def systems_by_errata(errata_uuids, installable, non_installable)
       end
 
       if installable
-        System.with_installable_errata(errata)
+        readable_systems.with_installable_errata(errata)
       elsif non_installable
-        System.with_non_installable_errata(errata)
+        readable_systems.with_non_installable_errata(errata)
       else
-        System.with_applicable_errata(errata)
+        readable_systems.with_applicable_errata(errata)
```

Comment:
I wouldn't bother creating a variable to hold readable_systems, just use  System.readable IMHO